src/ui/public/filter_manager/lib/phrase.js

```diff
@@ -17,8 +14,15 @@ export default function buildPhraseFilter(field, value, indexPattern) {
       convertedValue = value === 'true' ? true : false;
     }
 
+    // We must wrap painless scripts in a lambda in case they're more than a simple expression
+    let script = `(${field.script}) == value`;
+    if (field.lang === 'painless') {
+      script = `boolean compare(Supplier s, def v) {return s.get() == v;}
+                compare(() -> { ${field.script} }, params.value);`;
+    }
```

Comment:
Maybe this could be refactored?  `getInlineScriptForPhraseFilter`, or similar helper function on the field object, perhaps? Then it won't have to be written out three times.   I was toying with the idea when I first wrote this fix, but it was going in right before the release so I opted for minimal changes, but now there's a bit more content to be refactored so might be worth it.